apps/backoffice/src/app/api/auth/[...nextauth]/auth.ts

```diff
@@ -65,21 +68,90 @@ const retrieveApimUser = (config: Configuration) => (
   pipe(
     ApimUtils.getApimClient(config, config.AZURE_SUBSCRIPTION_ID),
     apimClient =>
-      ApimUtils.getApimService(
-        apimClient,
-        config.AZURE_APIM_RESOURCE_GROUP,
-        config.AZURE_APIM
-      ),
-    apimService =>
-      apimService.getUserByEmail(
-        `org.${identityTokenPayload.organization.id}@selfcare.io.pagopa.it` as EmailString,
-        true
+      pipe(
+        ApimUtils.getApimService(
+          apimClient,
+          config.AZURE_APIM_RESOURCE_GROUP,
+          config.AZURE_APIM
+        ),
+        apimService =>
+          apimService.getUserByEmail(
+            formatApimAccountEmailForSelfcareOrganization(
+              identityTokenPayload.organization
+            ),
+            true
+          ),
+        TE.mapLeft(
+          err =>
+            new Error(
+              `Failed to fetch user by its email, code: ${err.statusCode}`
+            )
+        ),
+        TE.chain(
+          flow(
+            O.fold(
+              () =>
+                pipe(
+                  TE.tryCatch(
+                    () =>
+                      apimClient.user.createOrUpdate(
+                        config.AZURE_APIM_RESOURCE_GROUP,
+                        config.AZURE_APIM,
+                        ulid(),
+                        {
+                          email: formatApimAccountEmailForSelfcareOrganization(
+                            identityTokenPayload.organization
+                          ),
+                          firstName: identityTokenPayload.organization.name,
+                          lastName: identityTokenPayload.organization.id,
+                          note: identityTokenPayload.organization.fiscal_code,
+                          appType: "backofficeIO"
+                        }
+                      ),
+                    E.toError
+                  ),
+                  TE.chain(apimUser =>
+                    TE.tryCatch(
+                      () =>
+                        Promise.all(
```

Comment:
this makes parallel calls, ensure that APi management can handle these without throttllng. more idiomatic: consider chain Taskeithers obtained using `TE.sequenceSeqArray`

Reply:
implemented with 7619510fce0ca50906382463aba09df4bfdda7e2